cherry_picker/cherry_picker/cherry_picker.py

```diff
@@ -422,10 +417,29 @@ def get_base_branch(cherry_pick_branch):
     """
     return '2.7' from 'backport-sha-2.7'
     """
-    prefix, sep, base_branch = cherry_pick_branch.rpartition('-')
+    prefix, sha, base_branch = cherry_pick_branch.split('-', 2)
+
+    if prefix != 'backport':
+        raise ValueError('branch name is not prefixed with "backport-".  Is this a cherry_picker branch?')
+    if not re.match('[0-9a-f]{7,40}', sha):
```

Comment:
Could also validate the sha against git (like it's done with a commit in config)

Reply:
Done.